conference.js

```diff
@@ -977,31 +1016,35 @@ export default {
                 this.videoSwitchInProgress = false;
                 this.toggleScreenSharing(false);
 
-                if(err === TrackErrors.CHROME_EXTENSION_USER_CANCELED)
+                if (err.name === TrackErrors.CHROME_EXTENSION_USER_CANCELED) {
                     return;
+                }
 
                 console.error('failed to share local desktop', err);
 
-                if (err === TrackErrors.FIREFOX_EXTENSION_NEEDED) {
+                if (err.name === TrackErrors.FIREFOX_EXTENSION_NEEDED) {
                     APP.UI.showExtensionRequiredDialog(
                         config.desktopSharingFirefoxExtensionURL
                     );
                     return;
                 }
 
                 // Handling:
+                // TrackErrors.PERMISSION_DENIED
                 // TrackErrors.CHROME_EXTENSION_INSTALLATION_ERROR
                 // TrackErrors.GENERAL
                 // and any other
-                let dialogTxt = APP.translation
-                    .generateTranslationHTML("dialog.failtoinstall");
-                let dialogTitle = APP.translation
-                    .generateTranslationHTML("dialog.error");
-                APP.UI.messageHandler.openDialog(
-                    dialogTitle,
-                    dialogTxt,
-                    false
-                );
+                let dialogTxt = APP.translation.generateTranslationHTML(
+                    err.name === TrackErrors.PERMISSION_DENIED
+                        ? "dialog.screenSharingPermissionDeniedError"
+                        : "dialog.failtoinstall");
+
+                let dialogTitle = APP.translation.generateTranslationHTML(
+                    err.name === TrackErrors.PERMISSION_DENIED
+                        ? "dialog.permissionDenied"
+                        : "dialog.error");
+
+                APP.UI.messageHandler.openDialog(dialogTitle, dialogTxt, false);
```

Comment:
The following should rely on the JS compiler supposedly optimizing the repeated err.name === TrackErrors.PERMISSION_DENIED, right?

```javascript
let dialogTxt, dialogTitle;
if (err.name === TrackErrors.PERMISSION_DENIED) {
  dialogTxt = "dialog.screenSharingPermissionDeniedError";
  dialogTitle = "dialog.permissionDenied";
} else {
...
}
APP.UI.messageHandler.openDialog(
  APP.translation.generateTranslationHTML(dialogTitle),
  APP.translation.generateTranslationHTML(dialogTxt),
  false);
```